File: Caper/CaperSharp/MapviewMappingEngine.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace CaperSharp
{
  class MapviewMappingEngine : MappingEngine
  {
    public MapviewMappingEngine( string aPath, Sequences aReferenceGenome )
      : base( aPath, aReferenceGenome )
    {
    }

    private int GetNormalizedIndex( string aValue )
    {
      return System.Convert.ToInt32( aValue ) - 1;
    }

    public override string GetSequence( string aLine )
    {
      return GetLinePieces( aLine )[ 14 ];
    }

    public override int GetIndex( string aLine )
    {
      return GetNormalizedIndex( GetLinePieces( aLine )[ 2 ] );
    }

    public override string GetContigIdent( string aLine )
    {
      return GetLinePieces( aLine )[ 1 ];
    }
  }
}

File: Caper/Caper/MapviewMappingEngine.cpp
#include "MapviewMappingEngine.h"

MapviewMappingEngine::MapviewMappingEngine(string aPath, Sequences * aGenome) 
: MappingEngine(aPath, aGenome), MapviewMappingUtilities()
{
}

MapviewMappingEngine::MapviewMappingEngine( char * aPath, SequenceEngine * lMappingEngine ) :
  MappingEngine(string(aPath), lMappingEngine->mSequences),
  MapviewMappingUtilities()
{
}


string MapviewMappingEngine::GetContigIdent( string & aLine )
{
  return MapviewMappingUtilities::GetContigIdent(aLine);
}

int MapviewMappingEngine::GetIndex( string & aLine )
{
  return MapviewMappingUtilities::GetIndex(aLine);
}

string MapviewMappingEngine::GetName( string & aLine )
{
  return MapviewMappingUtilities::GetName(aLine);
}

string MapviewMappingEngine::GetSequence( string & aLine )
{
  return MapviewMappingUtilities::GetSequence(aLine);
}

string MapviewMappingEngine::GetStrand( string & aLine )
{
  return MapviewMappingUtilities::GetStrand(aLine);
}